app/controllers/events_controller.rb

```diff
@@ -0,0 +1,27 @@
+class EventsController < ApplicationController
+  before_action :login_required, only: %i[new create]
+  def show
+    @event = Event.find(params[:id])
+    @event_creator = Event.creator(@event)
+  end
+
+  def index
+    @events = Event.all.limit(10)
+    @upcoming_events = @events.upcoming_date
+    @past_events = @events.past_date
+  end
+
+  def new
+    @event = Event.new
+  end
+
+  def create
+    @event = current_user.events.build(name: params[:name], body: params[:body], date: params[:date])
```

Comment:
For the code to be cleaner, you should write a private method that return these parameters (params[..)